final_task/rss_reader/rss_parser.py

```diff
@@ -0,0 +1,274 @@
+import sys
+import feedparser
+import json
+import logging
+import requests
+from yattag import Doc
+from fpdf import FPDF
+import os
+from database_functions import put_into_db, check_existance, json_from_cashe
+from database_functions import open_database, printing
+
+class RssParser():
+    """ Parses news from website"""
+    def __init__(self, *args):
+        """ Inits arguments"""
+
+        self.url = args[0]
+        self.limit = args[1]
+        self.jsonFormat = args[2]
+        self.date = args[3]
+        self.to_html = args[5]
+        self.path = args[4]
+        self.to_pdf = args[6]
+        logging.debug("Check in parser")
+        if self.date:
+            self.print_cashe()
+        else:
+            self.parse()
+
+    def print_cashe(self):
+        """ Prints news from database from the specified day."""
+        exists = check_existance()
+        if not exists:
+            print("There is no cache soon")
+            return None
+        con = open_database()
+
+        if con:
+            cur = con.cursor()
+            if self.limit > 0:
+                cur.execute("SELECT * from cache WHERE feed = ? AND date = ? LIMIT ?",
+                            (self.url, self.date, str(self.limit)))
+            else:
+                cur.execute("SELECT * from cache WHERE feed = ? AND date = ?", (self.url, self.date))
+
+            rows = cur.fetchall()
+            if not rows:
+                print("No results\nTry to enter another date or url")
+                return None
+            if self.jsonFormat:
+                json_from_cashe(rows)
+            elif self.to_html:
+                self.convert_into_html_format(rows)
+            elif self.to_pdf:
+                self.convert_into_pdf_format(rows)
+            else:
+                for row in rows:
+                    printing(row)
+            con.close()
+        else:
+            print("Can't connect to database")
+
+    def convert_into_html_format(self, thefeed):
+        """ Creates html file in specified directory and puts information in database, if it was read from website"""
+
+        # openning html-file to write
+        if not os.path.exists(self.path):
+            print("Invalid directory")
+            return None
+        filename = os.path.join(self.path, "news.html")
+        file = open(filename, 'w')
+        logging.debug("File was opened successfully")
+
+        doc, tag, text, line = Doc().ttl()
+        line('h1', 'News from ' + self.url)
+
+        for thefeedentry in thefeed:
```

Comment:
it is better to use `small_snake_case` for function, method and variable names  for better readability